azurelinuxagent/ga/exthandlers.py

```diff
@@ -993,7 +990,12 @@ def write_ext_handlers_status_to_info_file(self, vm_status):
             status.pop('formattedMessage', None)
             status.pop('substatus', None)
 
-        fileutil.write_file(status_path, json.dumps(data))
+        status_path = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE)
+        json_text = json.dumps(data)
+        fileutil.write_file(status_path, json_text)
+
+        if incarnation_changed:
+            GoalStateHistory(datetime.datetime.utcnow().isoformat()).save_status(json_text)
```

Comment:
The status is saved in a different directory with no suffix

Reply:
actually it'd be more useful to, on incarnation change, preserve the previous status file instead of overwriting (i.e. keep a history of the last status we reported for the previous goal state, rather than the first status of the new goal state).

will submit a change for this